src/mame/includes/tsconf.h

```diff
@@ -0,0 +1,212 @@
+#define SYS_CONFIG REG(0x20)
+#define MEM_CONFIG REG(0x21)
+#define HS_INT REG(0x22)
+#define VS_INT_L REG(0x23)
+#define VS_INT_H REG(0x24)
+
+#define INT_MASK REG(0x2a)
+
+#define T0_X_OFFSER_L REG(0x40)
+#define T0_X_OFFSER_H REG(0x41)
+#define T0_Y_OFFSER_L REG(0x42)
+#define T0_Y_OFFSER_H REG(0x43)
+#define T1_X_OFFSER_L REG(0x44)
+#define T1_X_OFFSER_H REG(0x45)
+#define T1_Y_OFFSER_L REG(0x46)
+#define T1_Y_OFFSER_H REG(0x47)
+
+static constexpr rgb_t rgb15(u16 rgb)
+{
+	u8 r = (BIT(rgb, 10, 5) << 3) | 0x07;
+	u8 g = (BIT(rgb, 5, 5) << 3) | 0x07;
+	u8 b = (BIT(rgb, 0, 5) << 3) | 0x07;
+	return rgb_t(r, g, b);
+}
+
+class tsconf_state : public spectrum_128_state
+{
+public:
+	tsconf_state(const machine_config &mconfig, device_type type, const char *tag)
+		: spectrum_128_state(mconfig, type, tag),
+		  m_bank1(*this, "bank1"), m_bank2(*this, "bank2"), m_bank3(*this, "bank3"), m_bank4(*this, "bank4"),
+		  m_keyboard(*this, "pc_keyboard"),
+		  m_beta(*this, BETA_DISK_TAG),
+		  m_sdcard(*this, "sdcard"),
+		  m_dma(*this, "dma"), m_cmos(*this, "cmos"), m_glukrs(*this, "glukrs"),
+		  m_gfxdecode(*this, "gfxdecode"),
+		  m_cram(*this, "cram")
+	{
+	}
+
+	void tsconf(machine_config &config);
+
+protected:
+	virtual void video_start() override;
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+
+private:
+	DECLARE_VIDEO_START(tsconf);
+
+	enum gluk_ext
+	{
+		CONF_VERSION = 0x00,
+		BOOTLOADER_VERSION = 0x01,
+		PS2KEYBOARDS_LOG = 0x02,
+		RDCFG = 0x03,
+		CONFIG = 0x0e,
+		SPIFL = 0x10,
+
+		DISABLED = 0xff
+	};
+
+	void tsconf_port_7ffd_w(u8 data);
+
+	u8 tsconf_port_xxaf_r(offs_t reg);
+	void tsconf_port_xxaf_w(offs_t reg, u8 data);
+
+	u8 tsconf_port_77_zctr_r(offs_t reg);
+	void tsconf_port_77_zctr_w(offs_t reg, u8 data);
+	u8 tsconf_port_57_zctr_r(offs_t reg);
+	void tsconf_port_57_zctr_w(offs_t reg, u8 data);
+	void tsconf_spi_miso_w(u8 data);
+
+	void tsconf_update_bank1();
+	u8 beta_neutral_r(offs_t offset);
+	u8 beta_enable_r(offs_t offset);
+	u8 beta_disable_r(offs_t offset);
+	template <unsigned Bank> void tsconf_bank_w(offs_t offset, u8 data);
+	void ram_bank_write(u8 bank, offs_t offset, u8 data);
+	void ram_page_write(u8 page, offs_t offset, u8 data);
+
+	void tsconf_io(address_map &map);
+	void tsconf_mem(address_map &map);
+	void tsconf_switch(address_map &map);
+
+	u16 ram_read16(offs_t offset);
+	void ram_write16(offs_t offset, u16 data);
+	u16 spi_read16();
+
+	required_memory_bank m_bank1;
+	required_memory_bank m_bank2;
+	required_memory_bank m_bank3;
+	required_memory_bank m_bank4;
+
+	required_device<at_keyboard_device> m_keyboard;
+
+	required_device<beta_disk_device> m_beta;
+	required_device<spi_sdcard_sdhc_device> m_sdcard;
+	required_device<tsconfdma_device> m_dma;
+
+	required_device<ram_device> m_cmos;
+	required_device<nvram_device> m_glukrs;
+	u8 tsconf_port_f7_cmos_r(offs_t offset);
+	void tsconf_port_f7_cmos_w(offs_t offset, u8 data);
+	gluk_ext m_gluk_ext;
+	u8 m_gluk_reg;
```

Comment:
Please don’t intermix data members and member functions within an access level.

Reply:
I tried to logically group them, but your point make sense as well. So will fix